### PR TITLE
Add integration test for parallel db access in middleware

### PR DESCRIPTION
**Pull request by myieye**
Make waiting for hg repos to be ready explicit

Add integration test for parallel db access in middleware

### CODE REVIEW

**Comment by myieye** [comment]
* **#1196** <a href="https://app.graphite.dev/github/pr/sillsdev/languageforge-lexbox/1196?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @myieye and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: backend/Testing/ApiTests/ApiTestBase.cs [hunk 1]
```diff
@@ -14,6 +14,7 @@ public class ApiTestBase
     public string BaseUrl => TestingEnvironmentVariables.ServerBaseUrl;
     private readonly SocketsHttpHandler _httpClientHandler;
     public readonly HttpClient HttpClient;
+    public string? CurrJwt { get; private set; }
```
Comment (hahn-kev) [inline backend/Testing/ApiTests/ApiTestBase.cs:17]: is this actually used anywhere? I don't see it

**Review by hahn-kev — COMMENTED** [review]
Looks good to me, but I had a question and didn't want this to auto merge.

edit: should this fail now because it doesn't include the hot chocolate fix?

File: backend/Testing/ApiTests/GqlMiddlewareTests.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        // our middlware will query the project confidentiality from the DB to determine
+        // if the user is allowed to view all members
+        var json = await QueryMyProjectsWithMembers();
+
+        json.ShouldNotBeNull();
+        var myProjects = json["data"]!["myProjects"]!.AsArray();
+        var ids = myProjects.Select(p => p!["id"]!.GetValue<Guid>());
+
+        foreach (var project in projects)
+            ids.ShouldContain(project.id);
```
Comment (hahn-kev) [inline backend/Testing/ApiTests/GqlMiddlewareTests.cs:73]:
```suggestion
ids.ShouldBeSubsetOf(projects.Select(p => p.id));
```

File: backend/Testing/ApiTests/ApiTestBase.cs (hunk 1 above)
Comment (myieye) [reply backend/Testing/ApiTests/ApiTestBase.cs:17]: The LexboxProject class reads.

**Comment by myieye** [comment]
> edit: should this fail now because it doesn't include the hot chocolate fix?

The HC fix is already in develop.

**Review by hahn-kev — APPROVED** [review]

**Comment by myieye** [comment]
### Merge activity

* **Nov 5, 4:43 AM EST**: [Graphite](https://app.graphite.dev/github/pr/sillsdev/languageforge-lexbox/1196) rebased this pull request as part of a merge.

**Pull request merged**